### ISSUE TITLE
[ANT-857] Invalid Thematic Trimming variable name "Profit" must be replaced with "Profit by plant"

### ISSUE THREAD

**Issue by laurent-laporte-pro** [AntaresSimulatorTeam/AntaREST#1741]
## Description

The variable used to display or hide power plant profits in the simulation output is named "Profit,"
which should be renamed to "Profit by plant" for accurate representation.

## Proposed Solution

Correct this variable in both the back end and front end by renaming it to "Profit by plant".

The unit tests should be updated to account for this change and test scenarios when the "Profit by plant"
feature is enabled or disabled.

## Steps to Reproduce

1. Choose a study in version 8.3 or above
2. Select the "CONFIGURATION" tab,
3. In the "Output profile" field set, choose "Thematic Trimming" and select "Custom",
4. Click on the gear button "⚙ Parameters" to customize the "Thematic Trimming",
5. In the "Thematic Trimming" dialog box, you can see the "Profit" variable.

## Environment

- Application Version: v2.14.5
- [x] Staging
- [x] Production

## Screenshots

![thematic_trimming_profit_by_plant](https://github.com/AntaresSimulatorTeam/AntaREST/assets/43534797/8cecbc3e-901e-4450-8efa-1151afef8e0e)

![thematic_trimming_profit_by_plant-2](https://github.com/AntaresSimulatorTeam/AntaREST/assets/43534797/957a23ae-455f-45aa-832b-9a0d2d192df2)

**Comment by laurent-laporte-pro** [comment]
There is no real risk to change the variable name from "Profit" to "Profit by plant" because:

- By default, all variables are enabled, which means that the `generaldata.ini` configuration file is empty because it represents the default behavior.

- However, if the user activates the "Profit" variable instead of "Profit by plant," an exclusion is set for "Profit." However, the simulator will raise an exception because it doesn't recognize this variable.

For instance:

```text
[2023-09-19 15:30:11][solver][warns] /path/to/study/settings/generaldata.ini: 'select_var +': Unknown property
[2023-09-19 15:30:38][solver][error] Found 1 warning: The simulation must stop.
```

In other words, there is no need to take backward compatibility measures in this case.
